regparser/tree/struct.py

```diff
@@ -134,6 +143,12 @@ def walk(node, fn):
     return results
 
 
+def filter_walk(node, fn):
+    """Perform fn for every node in the tree and return a list of nodes
+    on which the function returns truthy."""
+    return walk(node, lambda n: n if fn(n.label) else None)
```

Comment:
It doesn't look like your description matches the implementation? Specifically, it looks like you're only passing in the label rather than the whole node